### DISCUSSION TITLE
How to recconect old session

### DISCUSSION THREAD

**Discussion by amogsiddh** [Auties00/WhatsappWeb4j#343] [General]
i am reconnecting old session with phone number using following code.. please correct me if i am wrong.

Whatsapp wsp =
                Whatsapp
                        .webBuilder()
                        .serializer(new DefaultControllerSerializer(Path.of("C:\\whatsapp4j")))
                        .newConnection(phonenumber)
                        .name("WhatsAppVPN")
                        .historyLength(WebHistoryLength.ZERO)
                        .errorHandler(ErrorHandler.toTerminal())
                        .build()
                        .addListener((Listener) new Wsplistener())
                        .addDisconnectedListener(reason -> System.out.printf("Disconnected: %s%n", reason))
                        .connect()
                        .join();

it recoonect with that session.. but messages are not send.. please guide me.

**Comment by DanielSilvaol** [comment]
I'm using the UUID to reconnect, here's an example
```java
UUID id = "";
Whatsapp whatsapp = Whatsapp.webBuilder()
                .newConnection(id)
                .build()
                .addDisconnectedListener(onDisconnectedActions::onDisconnected)
                .addNewMessageListener(newMessageActions::onNewMessage)
                .connect()
                .get(120, TimeUnit.SECONDS);


whatsapp.store().uuid();
```

**Reply by devPedroFreire** [reply]
witch version are you using? at the actual its not possible to call build()

**Reply by Kronzii88** [reply]
could you give me the example for function onDisconnected

**Comment by Kronzii88** [comment]
is there any solution for this ?

**Reply by devPedroFreire** [reply]
did you find out?
